howso/client/api.py

```diff
@@ -0,0 +1,138 @@
+from __future__ import annotations
+
+from collections.abc import Mapping
+from functools import lru_cache
+import json
+from pathlib import Path
+import typing as t
+from uuid import uuid4
+
+from typing_extensions import NotRequired, TypeAlias, TypedDict
+
+from amalgam.api import Amalgam
+from .exceptions import HowsoError
+
+DEFAULT_ENGINE_PATH = Path(__file__).parent.parent.joinpath("howso-engine")
+
+SchemaTypeOption: TypeAlias = t.Literal["any", "assoc", "boolean", "list", "number", "string", "null"]
+"""Enum of valid Engine types."""
+
+SchemaType: TypeAlias = t.Union[SchemaTypeOption, list[SchemaTypeOption]]
+"""Valid schema type options."""
+
+TypeDefinition: TypeAlias = t.Union[SchemaType, "Ref", "Schema", "AnyOf"]
+"""Any valid type definition. Type(s), Reference, Schema, or Schema composition."""
+
+
+class Ref(TypedDict):
```

Comment:
Since these are all TypedDict and therefore the order of the keys isn't significant, how do you feel about alphabetizing them?

Reply:
Re-organized them a bit. Less useful I think strictly alphabetized. So I have grouped them based on required, usage then alphabetized.